server/tests/documents.spec.js

```diff
@@ -0,0 +1,285 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const app = require('../../app');
+const documentsHelper = require('./helpers/documentsHelper');
+const usersHelper = require('./helpers/userHelper');
+
+const dummyDocument = documentsHelper.documentsHelper;
+const dummyUser = usersHelper.usersHelper;
+
+const should = chai.should();
+chai.use(chaiHttp);
+
+describe('Documents', () => {
+  let userToken;
+  let adminToken;
+  before((done) => {
+    chai.request(app)
```

Comment:
This request (**line 17**) runs asynchronously and in parallel with the request made in **line 26**. `done` might be called while the former is still processing if the latter finishes before the former. The second request could probably be refactored and nested in the first request.